test/over_react/util/react_util_test.dart

```diff
@@ -21,19 +21,23 @@ import 'package:test/test.dart';
 
 main() {
   group('UiPropsMapView', () {
+    late Map backingMap;
     late UiPropsMapView mapView;
 
     setUp(() {
-      mapView = TestUiPropsMapView({
+      backingMap = {
         'id': 'test-id',
-      });
+      };
+      mapView = TestUiPropsMapView(backingMap);
     });
 
     test('reads from and writes to the backing map as expected', () {
       expect(mapView.id, 'test-id');
       mapView.id = 'something else';
       mapView['foo'] = 'bar';
       expect({...mapView}, {'id': 'something else', 'foo': 'bar'});
+      expect(backingMap, {...mapView},
+          reason: 'backing map should reflect all changes in the map view');
```

Comment:
The existing test didn't seem to verify that the backingMap was being updated; if `TestUiPropsMapView`/`UiPropsMapView` were making a copy of the backing map, the tests still would have passed. So I beefed up test coverage here to ensure that worked as expected.